File: content-solutions/backtracking/stair-steps.py
def solution(n):
    answer = []
    path = []

    def climb(remain):
        if remain == 0:
            answer.append(path[:])
            return
        for step in (1, 2):
            if step <= remain:
                path.append(step)
                climb(remain - step)
                path.pop()

    climb(n)
    return answer

File: content-solutions/backtracking/stair-steps.js
function solution(n) {
  const answer = [];
  const path = [];
  function climb(remain) {
    if (remain === 0) {
      answer.push([...path]);
      return;
    }
    for (const step of [1, 2]) {
      if (step <= remain) {
        path.push(step);
        climb(remain - step);
        path.pop();
      }
    }
  }
  climb(n);
  return answer;
}
